external_libraries/mpi_python/mpi_python.h

```diff
@@ -342,7 +429,19 @@ class PythonMPI
     template<class T>
     inline MPI_Datatype GetMPIDatatype(const T& Value);
 
-    int mArgc;
+    /// An auxiliary function to determine the MPI_Op corresponding to a given Enum type
+    /// This is necessary bcs MPI_Op cannot be directly exposed to Python
```

Comment:
Why not? (asking out of curiosity)

Reply:
Seems to be related to forward-declaration in mpi.h:
~~~
In file included from /usr/include/c++/5/bits/stl_iterator_base_types.h:67:0,
                 from /usr/include/c++/5/bits/forward_list.h:36,
                 from /usr/include/c++/5/forward_list:38,
                 from /home/philippb/software/Kratos_Master/external_libraries/pybind11/detail/common.h:139,
                 from /home/philippb/software/Kratos_Master/external_libraries/pybind11/pytypes.h:12,
                 from /home/philippb/software/Kratos_Master/external_libraries/pybind11/cast.h:13,
                 from /home/philippb/software/Kratos_Master/external_libraries/pybind11/attr.h:13,
                 from /home/philippb/software/Kratos_Master/external_libraries/pybind11/pybind11.h:43,
                 from /home/philippb/software/Kratos_Master/external_libraries/mpi_python/python_interface.cpp:18:
/usr/include/c++/5/type_traits: In instantiation of ‘struct std::is_base_of<pybind11::detail::pyobject_tag, ompi_op_t>’:
/home/philippb/software/Kratos_Master/external_libraries/pybind11/cast.h:1614:66:   required from ‘pybind11::object pybind11::cast(const T&, pybind11::return_value_policy, pybind11::handle) [with T = ompi_op_t*; typename std::enable_if<(! std::is_base_of<pybind11::detail::pyobject_tag, typename std::remove_reference< <template-parameter-1-1> >::type>::value), int>::type <anonymous> = 0]’
/home/philippb/software/Kratos_Master/external_libraries/pybind11/cast.h:1663:53:   required from ‘pybind11::object pybind11::detail::object_or_cast(T&&) [with T = ompi_op_t*; typename std::enable_if<(! std::is_base_of<pybind11::detail::pyobject_tag, typename std::remove_reference< <template-parameter-1-1> >::type>::value), int>::type <anonymous> = 0]’
/home/philippb/software/Kratos_Master/external_libraries/pybind11/pytypes.h:446:45:   required from ‘void pybind11::detail::accessor<Policy>::operator=(T&&) && [with T = ompi_op_t*; Policy = pybind11::detail::accessor_policies::str_attr]’
/home/philippb/software/Kratos_Master/external_libraries/mpi_python/python_interface.cpp:63:37:   required from here
/usr/include/c++/5/type_traits:1459:12: error: invalid use of incomplete type ‘struct ompi_op_t’
     struct is_base_of
            ^
In file included from /home/philippb/software/Kratos_Master/external_libraries/mpi_python/mpi_python.h:20:0,
                 from /home/philippb/software/Kratos_Master/external_libraries/mpi_python/python_interface.cpp:22:
/home/philippb/software/OpenMPI/openmpi-1.10.7_install/include/mpi.h:334:16: note: forward declaration of ‘struct ompi_op_t’
 typedef struct ompi_op_t *MPI_Op;

~~~

Reply:
this is from the try with `attr`
But also with `enum` it is not working